frontend/packages/db-structure/src/parser/prisma/parser.ts

```diff
@@ -34,28 +34,40 @@ async function parsePrismaSchema(schemaString: string): Promise<ProcessResult> {
       indices: {},
     }
   }
-
   for (const model of dmmf.datamodel.models) {
     for (const field of model.fields) {
-      if (
-        field.relationName &&
-        (field.relationFromFields?.length ?? 0) > 0 &&
+      if (!field.relationName) continue
+
+      const existingRelationship = relationships[field.relationName]
+      const isTargetField =
+        field.relationToFields?.[0] &&
         (field.relationToFields?.length ?? 0) > 0 &&
         field.relationFromFields?.[0] &&
-        field.relationToFields?.[0]
-      ) {
-        const relationship: Relationship = {
-          name: field.relationName,
-          primaryTableName: field.type,
-          primaryColumnName: field.relationToFields[0],
-          foreignTableName: model.name,
-          foreignColumnName: field.relationFromFields[0],
-          cardinality: 'ONE_TO_MANY',
-          updateConstraint: 'NO_ACTION',
-          deleteConstraint: 'NO_ACTION',
-        }
-        relationships[relationship.name] = relationship
-      }
+        (field.relationFromFields?.length ?? 0) > 0
+
+      const relationship: Relationship = isTargetField
+        ? {
+            name: field.relationName,
+            primaryTableName: field.type,
+            primaryColumnName: field.relationToFields[0] ?? '',
+            foreignTableName: model.name,
+            foreignColumnName: field.relationFromFields[0] ?? '',
+            cardinality: existingRelationship?.cardinality ?? 'ONE_TO_MANY',
+            updateConstraint: 'NO_ACTION',
+            deleteConstraint: 'NO_ACTION',
+          }
+        : {
+            name: field.relationName,
+            primaryTableName: existingRelationship?.primaryTableName ?? '',
+            primaryColumnName: existingRelationship?.primaryColumnName ?? '',
+            foreignTableName: existingRelationship?.foreignTableName ?? '',
+            foreignColumnName: existingRelationship?.foreignColumnName ?? '',
+            cardinality: field.isList ? 'ONE_TO_MANY' : 'ONE_TO_ONE',
+            updateConstraint: 'NO_ACTION',
+            deleteConstraint: 'NO_ACTION',
+          }
```

Comment:
`as const` might be good?


```suggestion
      const relationship = isTargetField
        ? {
            name: field.relationName,
            primaryTableName: field.type,
            primaryColumnName: field.relationToFields[0] ?? '',
            foreignTableName: model.name,
            foreignColumnName: field.relationFromFields[0] ?? '',
            cardinality: existingRelationship?.cardinality ?? 'ONE_TO_MANY',
            updateConstraint: 'NO_ACTION',
            deleteConstraint: 'NO_ACTION',
          } as const
        : {
            name: field.relationName,
            primaryTableName: existingRelationship?.primaryTableName ?? '',
            primaryColumnName: existingRelationship?.primaryColumnName ?? '',
            foreignTableName: existingRelationship?.foreignTableName ?? '',
            foreignColumnName: existingRelationship?.foreignColumnName ?? '',
            cardinality: field.isList ? 'ONE_TO_MANY' : 'ONE_TO_ONE',
            updateConstraint: 'NO_ACTION',
            deleteConstraint: 'NO_ACTION',
          } as const
```

Reply:
Thank you for your suggestion 👍
but I did not understand why `const` assertion would be better.
Because it can avoid rewriting of field?

Reply:
That's right! It might be readonly.

Reply:
Alright!
I've fixed 6f4bde0968d49aecd8425a80a88fb648f2cab4bc